src/librustc/middle/liveness.rs

```diff
@@ -1355,13 +1341,14 @@ impl<'a, 'tcx> Liveness<'a, 'tcx> {
         }
     }
 
-    fn propagate_through_loop(&mut self,
-                              expr: &Expr,
-                              kind: LoopKind<'_>,
-                              body: &hir::Block,
-                              succ: LiveNode)
-                              -> LiveNode {
+    fn propagate_through_loop(
+        &mut self,
+        expr: &Expr,
+        body: &hir::Block,
+        succ: LiveNode
+    ) -> LiveNode {
         /*
+        FIXME: clean up this description.
```

Comment:
N.B. this bit; I'm not entirely sure what the graph should say now...

This might be a reasonable interpretation:

```
        We model control flow like this:

              (expr) <--+
                |       |
                v       |
              (body) ---+
```

cc also @pnkfelix @nikomatsakis

Reply:
I would probably change `expr` to `loop expr` and add a note below saying something like "A `continue` expression targeting the loop will have a successor of `expr` and a `break` expression will have the successor of `expr` as its successor."